src/widgets/ot_pulltorefresh.eliom

```diff
@@ -0,0 +1,201 @@
+                    := Js.string "translateY(0)";
+                    refreshFlag := false))
+               700.)
+          (*if the page refreshing finishes instantaneously,
+              setTimeout is used to show the animation*)
+        else (
+          (*if page refresh fails*)
+          Conf.set_state @@ Some Failed;
+          js_container##.style##.transform := Js.string "translateY(0)";
+          ignore
+            (Dom_html.window##setTimeout
+               (Js.wrap_callback (fun () -> refreshFlag := false))
+               500.));
+        Lwt.return_unit)
+
+  let scroll_back () =
+    (*scroll back to top if |percentage| < dragThreshold*)
+    if !joinRefreshFlag
+    then (
+      Manip.Class.add container "ot-pull-refresh-transition-on";
+      js_container##.style##.transform := Js.string "translateY(0)";
+      ignore
+        (Dom_html.window##setTimeout
+           (Js.wrap_callback (fun () -> refreshFlag := false))
+           500.))
+
+  let touchend_handler ev _ =
+    if !percentage < 0. && !dragStart >= 0
+    then
+      if !refreshFlag
+      then Dom.preventDefault ev
+      else (
+        if -. !percentage > dragThreshold && !joinRefreshFlag
+        then refresh ()
+        else scroll_back ();
+        (*reinitialize paramaters*)
+        joinRefreshFlag := false;
+        dragStart := -1;
+        percentage := 0.);
+    Lwt.return_unit
+
+  let init () =
+    let open Js_of_ocaml_lwt.Lwt_js_events in
+    let ts = touchstarts js_container touchstart_handler in
+    let tm = touchmoves js_container touchmove_handler in
+    let te = touchends js_container touchend_handler in
+    let tc = touchcancels js_container touchend_handler in
+    [ts; tm; te; tc]
+end]
+
+let make ?(a = []) ?(dragThreshold = 0.3) ?(moveCount = 200)
+    ?(refresh_timeout = 20.) ?(header = [%shared default_header]) ~content
+    (afterPull : (unit -> bool Lwt.t) Eliom_client_value.t)
+  =
+  let state_s, set_state = Eliom_shared.React.S.create None in
+  let headContainer =
+    Eliom_content.Html.R.node
+    @@ Eliom_shared.React.S.map
+         [%shared
+           let open Eliom_content.Html in
+           fun s ->
+             F.div ~a:[F.a_class ["ot-pull-refresh-head-container"]]
+             @@ Eliom_shared.Value.local ~%header
+             @@ s]
+         state_s
+  in
+  let container =
+    div ~a:[a_class ["ot-pull-refresh-container"]] [headContainer; content]
+  in
+  ignore
+    [%client
+      (let module Ptr_conf = struct
+         let set_state = ~%set_state
+         let dragThreshold = ~%dragThreshold
+         let moveCount = ~%moveCount
+         let timeout = ~%refresh_timeout
+
+         let headContainerHeight () =
+           (To_dom.of_element ~%headContainer)##.scrollHeight
+
+         let container = ~%container
+         let afterPull = ~%afterPull
+       end
+       in
+       let module Ptr = Make (Ptr_conf) in
+       let unlisten = ref (fun () -> ()) in
```

Comment:
Could you just call `Ptr.init ()` right away and be done with that?

Reply:
I could. No time today, but I'll do that asap.